sql/core/src/test/scala/org/apache/spark/sql/execution/command/v2/DropNamespaceSuite.scala

```diff
@@ -0,0 +1,33 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.execution.command.v2
+
+import org.apache.spark.SparkException
+import org.apache.spark.sql.execution.command
+
+/**
+ * The class contains tests for the `DROP NAMESPACE` command to check V2 table catalogs.
+ */
+class DropNamespaceSuite extends command.DropNamespaceSuiteBase with CommandSuiteBase {
+  override protected def assertDropFails(): Unit = {
+    val e = intercept[SparkException] {
```

Comment:
Can we leave a TODO to unify the error? I think both v1 and v2 commands should throw `AnalysisException`, and the error message should only have one difference between v1 and v2: `Cannot drop a non-empty namespace: ...` vs `Cannot drop a non-empty database: ...`

Reply:
If you think the change is small, you can also include it in this PR

Reply:
I will create a follow-up PR for this, it's ok? @cloud-fan

Reply:
> Can we leave a TODO to unify the error? I think both v1 and v2 commands should throw `AnalysisException`, and the error message should only have one difference between v1 and v2: `Cannot drop a non-empty namespace: ...` vs `Cannot drop a non-empty database: ...`

The error message of Hive External Catalog test is `Database ns is not empty. One or more tables exist.`, which is identical to the error message of v1. So we should keep this and remove TODO, or we should unify the error of v1, v2 test and then seperate the hive's one. Thanks. @cloud-fan @huaxingao

Reply:
can we change the v1 error message to match v2?

Reply:
yes, we can. After that, the test in `Hive External Catalog` would fail since it extends `v1.DropNamespaceSuiteBase`. So I think we should unify the error of v1, v2 test, and then create another one in Hive external catalog test that contains error message `Database ns is not empty. One or more tables exist.`. WDYT, @cloud-fan ?

Reply:
can we unify the error message everywhere? We can try-catch the error from hive catalog and re-throw it with a different error message?

Reply:
yeah, I think we can. I'll try. Thank you, @cloud-fan. :smiley: